### DISCUSSION TITLE
Conflicting Overloads and Redeclaration Errors in Eclipse

### DISCUSSION THREAD

**Discussion by nathanspectacular** [pmd/pmd#3452] [Q&A]
I created a new workspace in Eclipse 2021-06.  I cloned the PMD repository.  There are two projects that have errors: pmd-java and pmd-modelica.  Here are the errors in pmd-java.

![image](https://user-images.githubusercontent.com/88563519/128647412-13c35b55-819f-4262-9c15-030008e74e46.png)

I have Kotlin 0.8.21.v202008.16-2059 installed in Eclipse.  Running `mvnw.cmd clean verify` on the Windows 10 command line works perfectly so I know it is a problem with my Eclipse configuration.

When I go to Open Resource and enter `ASTCatchStatementTest`, I get the following matches.  It seems like they are all the same file.

![image](https://user-images.githubusercontent.com/88563519/128727408-ada5605d-797a-4b4c-8863-7478b8cee8b3.png)

How do I fix these errors?

**Comment by adangel** [comment]
Kotlin support in eclipse is unfortunately in the state of "not working". If I need to change the Kotlin test sources, I use IntelliJ.
In Eclipse, I make sure to close the projects, which contain Kotlin sources (especially pmd-lang-test), so that the jar is used instead. And I do *not* use the Kotlin plugin. It just doesn't work.

So, If you want to modify ASTCatchStatementTest.kt - either use a Text editor and compile in via maven or use IntelliJ.

Let me know, if you find a way, which makes eclipse+kotlin usable.

**Reply by nathanspectacular** [reply]
I closed all of the projects that had Kotlin except for the pmd-java project.  I want to add rules to this project.  Unfortunately, the pmd-java project has 10 Kotlin files in the src/test/kotlin/net/sourceforge/pmd/lang/java/ast directory.
